test/test_indexing.py

```diff
@@ -1328,6 +1328,17 @@ def test_gather_take_along_dim_cross_device(self, device, dtype):
                                     r"Expected tensor to have .* but got tensor with .* torch.take_along_dim()"):
             torch.take_along_dim(t.cpu(), indices, dim=0)
 
+    @onlyCUDA
+    def test_cuda_broadcast_index_use_deterministic_algorithms(self, device):
+        torch.use_deterministic_algorithms(True)
```

Comment:
Please use the `DeterministicGuard` context manager in `torch/testing/_internal/common_utils.py`, instead of changing the global determinism setting directly. If this test ever fails or quits earlier than expected, the context manager will ensure that determinism gets set back to its original setting

Reply:
Thank you for your advice, I have fixed it!